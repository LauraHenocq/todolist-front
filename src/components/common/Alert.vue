<script setup lang="ts">
import AlertType from '@/types/alert-type.ts';

type Props = {
  type: AlertType
  title: string
  text: string
}

defineProps<Props>()

const emit = defineEmits<{
  (e: 'close'): void;
}>();

const handleClose = () => {
  emit('close');
}

</script>
<template>
  <v-alert
    data-cy="alert"
    class="alert"
    :type="type"
    :title="title"
    :text="text"
    closable
    @click:close="handleClose"
  ></v-alert>
</template>
<style scoped>
.alert {
  position: fixed;
  bottom: 20px;
  right: 20px;
  left: 20px;
  z-index: 1;
}
</style>