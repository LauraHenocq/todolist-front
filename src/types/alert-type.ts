export const enum AlertType {
  success = 'success',
  warning = 'warning',
  error = 'error',
  info = 'info'
}