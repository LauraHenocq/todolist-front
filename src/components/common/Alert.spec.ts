import { mount, VueWrapper } from '@vue/test-utils';
import { nextTick } from 'vue';
import { beforeEach, expect, it, describe } from 'vitest'
import { createVuetify } from 'vuetify';
import Alert from './Alert.vue';
import { AlertType } from '@/types/alert-type';

const vuetify = createVuetify();

describe('TaskCard', () => {
  let wrapper: VueWrapper

  const title = 'Titre de l\'alerte';
  const text = 'Description de de l\'alerte';
  const type = AlertType.success;

  const defaultProps = {
    title,
    text,
    type
  }

  beforeEach(() => {
    wrapper = mount(Alert, {
      props: defaultProps,
      global: {
        plugins: [vuetify],
      },
    });
  });

  describe('render', () => {
    it('should render', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('actions', () => {
    it('should emit close event when the close element of the alert is clicked', async () => {
      (wrapper
        .findComponent('[data-cy="alert"]') as VueWrapper)
        .vm.$emit('click:close');

      await nextTick();

      expect(wrapper.emitted('close')).toHaveLength(1);
    });
  });
});