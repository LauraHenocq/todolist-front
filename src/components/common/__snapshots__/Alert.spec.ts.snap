// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskCard > render > should render 1`] = `
"<div data-v-c992c02f="" class="v-alert v-theme--light bg-success v-alert--density-default v-alert--variant-flat alert" role="alert" data-cy="alert">
  <!----><span class="v-alert__underlay"></span>
  <!---->
  <div class="v-alert__prepend"><i class="mdi-check-circle mdi v-icon notranslate v-theme--light" style="font-size: 28px; height: 28px; width: 28px;" aria-hidden="true" density="default"></i></div>
  <div class="v-alert__content">
    <div class="v-alert-title">Titre de l'alerte</div>Description de de l'alerte
    <!---->
  </div>
  <!---->
  <div class="v-alert__close"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-text" aria-label="Close"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
      <!---->
      <!---->
    </button></div>
</div>"
`;
